src/main/java/org/prebid/server/auction/AuctionRequestFactory.java

```diff
@@ -490,4 +499,68 @@ BidRequest validateRequest(BidRequest bidRequest) {
         }
         return bidRequest;
     }
+
+    /**
+     * Returns filled out {@link AuctionContext} based on given arguments.
+     * <p>
+     * Note: {@link TimeoutResolver} used here as argument because this method is utilized in AMP processing.
+     */
+    Future<AuctionContext> toAuctionContext(RoutingContext routingContext, BidRequest bidRequest,
+                                            long startTime, TimeoutResolver timeoutResolver) {
+        final Timeout timeout = timeout(bidRequest, startTime, timeoutResolver);
+
+        return accountFrom(bidRequest, timeout)
+                .map(account -> AuctionContext.builder()
+                        .routingContext(routingContext)
+                        .uidsCookie(uidsCookieService.parseFromRequest(routingContext))
+                        .bidRequest(bidRequest)
+                        .timeout(timeout)
+                        .account(account)
+                        .build());
+    }
+
+    /**
+     * Returns {@link Timeout} based on request.tmax and adjustment value of {@link TimeoutResolver}.
+     */
+    private Timeout timeout(BidRequest bidRequest, long startTime, TimeoutResolver timeoutResolver) {
+        final long timeout = timeoutResolver.adjustTimeout(bidRequest.getTmax());
+        return timeoutFactory.create(startTime, timeout);
+    }
+
+    /**
+     * Returns {@link Account} fetched by {@link ApplicationSettings}.
+     */
+    private Future<Account> accountFrom(BidRequest bidRequest, Timeout timeout) {
+        final String accountId = accountIdFrom(bidRequest);
+
+        return applicationSettings.getAccountById(accountId, timeout)
```

Comment:
accountIdFrom can return empty String, I propose to do check before applicationSettings.getAccountById call, as we do not check it in getAccountById implementations